src/main/java/soot/FastHierarchy.java

```diff
@@ -113,16 +113,8 @@ public Interval(int lower, int upper) {
     }
 
     public boolean isSubrange(Interval potentialSubrange) {
-      if (potentialSubrange == null) {
-        return false;
-      }
-      if (lower > potentialSubrange.lower) {
-        return false;
-      }
-      if (upper < potentialSubrange.upper) {
-        return false;
-      }
-      return true;
+      return (potentialSubrange == this)
```

Comment:
Here I converted the original checks into a single line and added the (potentialSubrange == this) check. This check implies the original (i.e. if potentialSubrange == this, then the lower and upper fields are equal) but is faster so I added it as an initial fast path.